openedx/features/enterprise_support/tests/test_api.py

```diff
@@ -90,8 +94,41 @@ def setUpTestData(cls):
             email='ent_worker@example.com',
             password='password123',
         )
+        cls.enterprise_customer = EnterpriseCustomerFactory()
+        cls.pending_enterprise_customer_user = PendingEnterpriseCustomerUserFactory()
         super().setUpTestData()
 
+
+    @mock.patch('openedx.features.enterprise_support.api.create_jwt_for_user')
+    def test_list_learners_filtered(self, mock_jwt_builder):
+        """
+        Test that the list learners endpoint can be filtered by user details
+        """
+        user = UserFactory(email='foobar@example.com')
+        UserProfileFactory.create(
+            user=user,
+            defaults={'name': 'Test Enterprise User'},
+        )
+        group = EnterpriseGroupFactory(
+            enterprise_customer=self.enterprise_customer,
+        )
+        pending_user = PendingEnterpriseCustomerUserFactory(
+            user_email="foobar@example.com",
+            enterprise_customer=self.enterprise_customer,
+        )
+
+        pending_user_query_string = f'?user_query={pending_user.user_email}'
+        # approach 1
+        api_client = self._assert_api_client_with_user(EnterpriseApiClient, mock_jwt_builder)
+        response = api_client.get_learners(group.uuid, pending_user_query_string)
+        assert response.status_code == 200
+
+        # approach 2
+        url = f'https://testserver/enterprise/api/v1/enterprise-group/{group.uuid}/learners/{pending_user_query_string}'
+        response = self.client.get(url)
+        assert response.status_code == 200
```

Comment:
error msg: `where 404 = <HttpResponseNotFound status_code=404, "text/html; charset=utf-8">.status_code`

Reply:
ok, for this one, you'll want to programatically compute what the url will be using Django's `reverse()` function.
https://docs.djangoproject.com/en/5.1/ref/urlresolvers/#reverse
Using reverse means you have to know the full view name of, in this case, the enterprise group view.  In general to figure out the view name, you start from the top-most `urls.py` file and work your way down.  So since enterprise is installed into the `lms` django project, you'd start here: https://github.com/openedx/edx-platform/blob/1a8c1e1c5da363c952c6f99f7c54c3a10c709b1e/lms/urls.py#L871-L875
From this one, we know the full view name starts with `enterprise`.
and realize you need to next go to the enterprise `urls.py` [file](https://github.com/openedx/edx-enterprise/blob/12a552b1d539e6098da98c514825234682397c4a/enterprise/urls.py#L24)
From this one, the next part of the view name is `api`.
Eventually you'll work your way down to here: https://github.com/openedx/edx-enterprise/blob/12a552b1d539e6098da98c514825234682397c4a/enterprise/api/v1/urls.py#L191-L196
So we know we need `v1.enterprise-group-learners` as the last part of the view name.
So in total, you'll want to compute `url` in this approach as
```python
url = reverse('enterprise.api.v1.enterprise-group-learners')
```